src/main/java/pl/javastart/restoffers/category/CategoryService.java

```diff
@@ -0,0 +1,42 @@
+package pl.javastart.restoffers.category;
+
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    public CategoryService(CategoryRepository categoryRepository, CategoryMapper categoryMapper) {
+        this.categoryRepository = categoryRepository;
+        this.categoryMapper = categoryMapper;
+    }
+
+    public List<CategoryDto> getListOfDtoCategories() {
+        List<CategoryDto> categoryDtoList = new ArrayList<>();
+        categoryRepository.findAll().forEach(category -> categoryDtoList.add(categoryMapper.convertToDtoCategory(category)));
+        return categoryDtoList;
+    }
+
+    public List<String> listOfCategoryNames() {
+        List<String> categoryNames = new ArrayList<>();
+        categoryRepository.findAll().forEach( category -> categoryNames.add(category.getName()));
```

Comment:
Lepiej wykonać zapytanie po stronie bazy danych, które będzie projekcją na same nazwy. W tabeli z kategoriami może być potencjalnie wiele rzeczy, których tutaj nie potrzebujemy. Filtrowanie po stronie kodu Javy to w przypadku baz danych zawsze gorsza opcja